.github/workflows/lint-test.yml

```diff
@@ -0,0 +1,26 @@
+name: 'Lint and test'
+
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
+    branches:
+      - 'main'
+  workflow_dispatch:
```

Comment:
If CI will guard every PR, curious the reason we still need workflow_dispatch here.

Reply:
Not sure, but maybe somebody will want to lint a branch that's not part of a PR, and I think this will let them do that.